src/borg/archive.py

```diff
@@ -1877,8 +1877,11 @@ def valid_item(obj):
                 return
         num_archives = len(archive_infos)
 
+        pi = ProgressIndicatorPercent(total=num_archives, msg='Checking archives %3.1f%%', step=0.1,
+                                      msgid='archive.robust_iterator')
```

Comment:
0.1 looks suspicious as num_archives is an integer.

Reply:
0.1 is what is used in other places with integer "input" counts and while the documentation of the `step` parameter in `ProgressIndicatorPercent` is rather vague, it does state it to be in prcent and the code does treat it as steps of the percentage output:
https://github.com/borgbackup/borg/blob/7a872bbcddc1e4eccf9cfabaefa2a76d57c68582/src/borg/helpers/progress.py#L116
https://github.com/borgbackup/borg/blob/7a872bbcddc1e4eccf9cfabaefa2a76d57c68582/src/borg/helpers/progress.py#L134

So in this case, output is displayed, whenever the progress percentage increased by 0.1% or more since the last invocation of `show()`. This does not hurt with few archives either, because updates happen no more than one archive processed (and `show()` invoked) anyway.

Reply:
Right, makes sense.